src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs

```diff
@@ -0,0 +1,356 @@
+
+                reason = OutOfRange;
+
+                return false;
+            }
+
+            if (targetingSettings.Audience.Groups != null)
+            {
+                int index = 0;
+
+                foreach (GroupRollout groupRollout in targetingSettings.Audience.Groups)
+                {
+                    index++;
+
+                    if (groupRollout.RolloutPercentage < 0 || groupRollout.RolloutPercentage > 100)
+                    {
+                        //
+                        // Audience.Groups[1].RolloutPercentage
+                        paramName = $"{targetingSettings.Audience}.{targetingSettings.Audience.Groups}[{index}].{groupRollout.RolloutPercentage}";
+
+                        reason = OutOfRange;
+
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines if a given context id should be targeted based off the provided percentage range
+        /// </summary>
+        public static bool IsTargeted(ITargetingContext targetingContext, double from, double to, string seed, bool ignoreCase, string hint)
+        {
+            string userId = ignoreCase ?
+                targetingContext.UserId.ToLower() :
+                targetingContext.UserId;
+
+            string contextId;
+
+            if (!string.IsNullOrEmpty(seed))
+            {
+                contextId = $"{userId}\n{seed}";
+            }
+            else
+            {
+                contextId = $"{userId}\n{hint}";
+            }
+
+            return IsTargeted(contextId, from, to);
+        }
+
+        /// <summary>
+        /// Determines if a given context id should be targeted based off the provided percentage
+        /// </summary>
+        /// <param name="contextId">A context identifier that determines what the percentage is applicable for</param>
+        /// <param name="from">The lower bound of the percentage for which the context identifier will be targeted</param>
+        /// <param name="to">The upper bound of the percentage for which the context identifier will be targeted</param>
+        /// <returns>A boolean representing if the context identifier should be targeted</returns>
+        private static bool IsTargeted(string contextId, double from, double to)
+        {
+            byte[] hash;
+
+            using (HashAlgorithm hashAlgorithm = SHA256.Create())
+            {
+                hash = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes(contextId));
+            }
+
+            //
+            // Use first 4 bytes for percentage calculation
+            // Cryptographic hashing algorithms ensure adequate entropy across hash
+            uint contextMarker = BitConverter.ToUInt32(hash, 0);
+
+            double contextPercentage = (contextMarker / (double)uint.MaxValue) * 100;
+
+            return contextPercentage >= from && contextPercentage < to;
```

Comment:
Because this is now `<` instead of `<=` we might need to edge case 100%. Which I think has to be something like:

```
if (to == 100%) {
  contextPercentage <= to
} else {
  contextPercentage < to
}
```

Or you could do:

```
double contextPercentage = (uint.MaxValue / ((double)uint.MaxValue + 1)) * 100;
```

So 100 is not an exact option. I think I prefer the +1 approach ^

Reply:
We'll handle this in a different PR.

Reply:
Given that we are touching this code, and also modifying it I think we need to handle this now.

I suggest this

```
//
// Handle edge case of exact 100 bucket
if (contextPercentage == 100)
{
    return to == 100;
}
```

Reply:
https://github.com/microsoft/FeatureManagement-Dotnet/pull/251 introduced the edge case. You can include the adjusted logic here. Could do:
```
//
// Handle edge case of exact 100 bucket
if (to == 100) {
  return contextPercentage >= from;
}

return contextPercentage >= from && contextPercentage < to;
```

And remove the check from the beginning.